packages/build/src/plugins_core/dev_blobs_upload/index.ts

```diff
@@ -0,0 +1,18 @@
+import { uploadBlobs } from '../blobs_upload/index.js'
+import { type CoreStep, type CoreStepCondition } from '../types.js'
+
+const condition: CoreStepCondition = async (...args) => {
+  const {
+    constants: { IS_LOCAL },
+  } = args[0]
+  return IS_LOCAL && ((await uploadBlobs.condition?.(...args)) ?? true)
+}
+
+export const devUploadBlobs: CoreStep = {
+  event: 'onDev',
+  coreStep: uploadBlobs.coreStep,
```

Comment:
Maybe I should've made that clearer, but what I meant by "reuse common functionality" https://github.com/netlify/build/pull/5552#discussion_r1535297826 was moving shared logic into a separate file that both core steps would use.

In this case, you're importing one core step into the other and reusing the logic it implements. There's no issue with that functionality-wise, but I worry that future us will be changing one core step unaware that it's actually affecting another core step that imports it.

I'm approving because I don't want to block you, but I'd recommend revisiting this if possible.

Reply:
I ended up splitting some more common functionality out into the `utils/blobs.ts` file and otherwise duplicated most of the contents of `upload_blobs` to `dev_upload_blobs`.

In general I share your concern, but in practice, it seems like the two steps should behave identically right now. Given that, I'm not sure how best to split out any more common functionality without simply moving the entire core step into a shared file.

I'm going to merge this as it stands so we can address the issue this solves, but if you have suggestions on how we might better structure this, let me know--happy to address it in a follow-up PR!